website/code_issues.qmd

```diff
@@ -128,8 +128,91 @@ foo <- function(..., verbose = TRUE){
 
 Functions can print per default, like the example above, as long as the printing can be turned off (here, by setting `verbose = FALSE`).
 
+:::{.callout-note}
+
+`print()` and `cat()` are not the only functions which can write output onto the console. The issue described in the recipe, also applies to the use of other printing function like `writeLines()`. 
+If you are using loggers to document your functions' process, make sure that users can set their log level such that not messages are displayed.
+
+:::
+
 
 
 ---------------------------------------------------------
 
-# Change of options() and par() 
\ No newline at end of file
+# Change of Options, Graphical Parameters and Working Directory
+
+## Problem
+
+You are changing the `par()`, `options()` or `setwd()` in your functions, examples, demos or vignettes without resetting them.
```

Comment:
I think this should be expanded to `Sys.setenv()` ans `Sys.setLanguage()` among other persistent environment that can alter the R output.

Reply:
There are no plans on a `Sys.setenv()` and `Sys.setLanguage()` section. And in my two years of reviewing CRAN packages I never came across a package which change them in the first place. I will add a line that this recipe not only applies to `par()`, `options()` and `setwd()` but all other functions which change parameters persitently.

Reply:
I know that forcats faced a failing test because one of the machines was in German and the order of some factors changed in the languages. This kind of things can make package maintainers stumble, mentioning them will help anyone depending on them or changing  them to be aware.